plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -6535,7 +6535,10 @@ private HashMap<String, VmStatsEntry> getVmStats(List<String> vmNames) throws Ex
                         }
                     }
 
-                    final VmStatsEntry vmStats = new VmStatsEntry(0, NumberUtils.toDouble(memkb) * 1024, NumberUtils.toDouble(guestMemusage) * 1024, NumberUtils.toDouble(memlimit) * 1024,
+                    double doubleMemKb = NumberUtils.toDouble(memkb)
+                    double guestFreeMem =  doubleMemKb - NumberUtils.toDouble(guestMemusage);
+
+                    final VmStatsEntry vmStats = new VmStatsEntry(0, doubleMemKb * 1024, guestFreeMem * 1024, NumberUtils.toDouble(memlimit) * 1024, NumberUtils.toDouble(memlimit) * 1024,
```

Comment:
if I count correctly, you are replacing 4 for 5 parameters to the constructor on VmStatsEntry. It takes 13 and now has 14 in the call. I don´t see a change to the constructor itself.

Reply:
I'm sorry, I probably made a mistake when cherry-picking this change.